[PyTorch / Sigrid GPU] Fixes in pinned stats collection and add new ODS pinned memory stats

banitag1 opened the pull request:
We do some fixes in pinned memory allocation stats collection and better differentiate between active vs allocated bytes.
Reviewed By: bbus, sayitmemory

Differential Revision: D83162346

banitag1 commented:
@pytorchbot label "topic: not user facing"

mradmila reviewed (approved):
I only reviewed the changes wrt to the HostStats and how we collect those.

banitag1 committed:
[PyTorch / Sigrid GPU] Fixes in pinned stats collection and add new ODS pinned memory stats (#164412)

Summary:

We do some fixes in pinned memory allocation stats collection and better differentiate between active vs allocated bytes.

Reviewed By: bbus, mradmila, sayitmemory

Differential Revision: D83162346
